core/modules/node/node.module

```diff
@@ -2268,6 +2272,16 @@ function node_page_default() {
  * @see node_menu()
  */
 function node_page_view(Node $node) {
+  // Determine if user has permission to view full page directly.
+  // Display "page not found" if not.
+  global $user;
+  $pageless = config_get('node.type.' . $node->type, 'settings.pageless');
+  $bypass_pageless = user_access('access page-less content', $user);
```

Comment:
`user_access()` second parameter is default to `global $user`, so code could be lighter here.

```
$pageless = config_get('node.type.' . $node->type, 'settings.pageless');
if ($pageless && !user_access('access page-less content')) {
```